src/group_theory/perm/list.lean

```diff
@@ -0,0 +1,337 @@
+/-
+Copyright (c) 2021 Yakov Pechersky. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yakov Pechersky
+-/
+
+import data.list.rotate
+import group_theory.perm.support
+
+/-!
+# Permutations from a list
+
+A list `l : list α` can be interpreted as a `equiv.perm α` where each element in the list
+is permuted to the next one, defined as `form_perm`. When we have that `nodup l`,
+we prove that `equiv.perm.support (form_perm l) = l.to_finset`, and that
+`form_perm l` is rotationally invariant, in `form_perm_rotate`.
+
+-/
+
+namespace list
+
+variables {α β : Type*}
+
+section form_perm
+
+variables [decidable_eq α] (l : list α)
+
+open equiv equiv.perm
+
+/--
+A list `l : list α` can be interpreted as a `equiv.perm α` where each element in the list
+is permuted to the next one, defined as `form_perm`. When we have that `nodup l`,
+we prove that `equiv.perm.support (form_perm l) = l.to_finset`, and that
+`form_perm l` is rotationally invariant, in `form_perm_rotate`.
```

Comment:
If an element appears twice in the list, what does the resulting `form_perm` do? I'm sure this is the case you don't care about, especially because of the `nodup`s everywhere, but it could be nice to say something about this, even if it's just that `form_perm` is intended to be used on `nodup` lists